keps/sig-network/2091-cluster-network-policy/README.md

```diff
@@ -0,0 +1,1161 @@
+
+n/a
+
+## Implementation History
+
+- 2021-02-18 - Created initial PR for the KEP
+
+## Drawbacks
+
+Securing traffic for a cluster for administrator's use case can get complex.
+This leads to introduction of a more complex set of APIs which could confuse
+users.
+
+## Alternatives
+
+Following alternative approaches were considered:
+
+### NetworkPolicy v2
+
+A new version for NetworkPolicy, v2, was evaluated to address features and use cases
+documented in this KEP. Since the NetworkPolicy resource already exists, it would be
+a low barrier to entry and can be extended to incorporate admin use cases.
+However, this idea was rejected because the NetworkPolicy resource was introduced
+solely to satisfy a developers intent. Thus, adding new use cases for a cluster admin
+would be contradictory. In addition to that, the administrator use cases are mainly
+scoped to the cluster as opposed to the NetworkPolicy resource, which is `namespaced`.
+
+### Empower, Deny, Allow action based CRD
+
+Unlike the NetworkPolicy resource in which each rule represents an allowed
+traffic, ClusterNetworkPolicy will enable administrators to set `Empower`,
+`Deny` or `Allow` as the action of each rule. ClusterNetworkPolicy rules should
+be read as-is, i.e. there will not be any implicit isolation effects for the Pods
+selected by the ClusterNetworkPolicy, as opposed to what NetworkPolicy rules imply.
+
+In terms of precedence, the aggregated `Empower` rules (all ClusterNetworkPolicy
+rules with action `Empower` in the cluster combined) should be evaluated before
+aggregated ClusterNetworkPolicy `Deny` rules, followed by aggregated ClusterNetworkPolicy
+`Allow` rules, followed by NetworkPolicy rules in all Namespaces. As such, the
+`Empower` rules have the highest precedence, which shall only be used to provide
+exceptions to deny rules. The `Empower` rules do not guarantee that the traffic
+will not be dropped: it simply denotes that the packets matching those rules can bypass the
+ClusterNetworkPolicy `Deny` rule evaluation.
+
+ClusterNetworkPolicy `Deny` rules are useful for administrators to explicitly
+block traffic from malicious clients, or workloads that poses security risks.
+Those traffic restrictions can only be lifted once the `Deny` rules are deleted
+or modified by the admin. In clusters where the admin requires total control over
+security postures of all workloads, the `Deny` rules can also be used to deny all
+incoming/outgoing traffic in the cluster, with few exceptions that's listed out
+by `Empower` rules.
+
+On the other hand, the `Allow` rules can be used to call out traffic in the cluster
+that needs to be allowed for certain components to work as expected (egress to
+CoreDNS for example). Those traffic should not be blocked when developers apply
+NetworkPolicy to their Namespaces which isolates the workloads.
+
+#### ClusterDefaultNetworkPolicy resource
+
+A ClusterDefaultNetworkPolicy resource will help the administrators set baseline
```

Comment:
likewise

(you probably don't need all the detail either)

Reply:
Done